anndata/utils.py

```diff
@@ -223,3 +223,30 @@ def is_deprecated(attr):
             for item in type.__dir__(cls)
             if not is_deprecated(getattr(cls, item, None))
         ]
+
+
+def import_function(module: str, name: str):
+    """\
+    Try to import function from module. If the module is not installed or
+    function is not part of the module, it returns a dummy function that raises
+    the respective import error once the function is called. This could be a
+    ModuleNotFoundError if the module is missing or an AttributeError if the
+    module is installed but the function is not exported by it.
+
+    Params
+    -------
+    module
+        Module to import from. Can be nested, e.g. "sklearn.utils".
+    name
+        Name of function to import from module.
+    """
+    try:
+        module = __import__(module, fromlist=[name])
```

Comment:
```suggestion
    """
    from importlib import import_module 
    try:
        module = import_module(module)
```

From the docs for `__import__`, I think this might be the way to go.

```
__import__(name, globals=None, locals=None, fromlist=(), level=0) -> module

Import a module. Because this function is meant for use by the Python
interpreter and not for general use, it is better to use
importlib.import_module() to programmatically import a module.
```